File: src/test/java/no/publishers/TestData.java
package no.publishers;

import com.google.common.collect.ImmutableMap;
import no.publishers.generated.model.PrefLabel;
import no.publishers.generated.model.Publisher;
import no.publishers.model.PublisherDB;
import org.bson.types.ObjectId;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Map;

public class TestData {
    private static final String MONGO_USER = "testuser";
    private static final String MONGO_PASSWORD = "testpassword";
    private static final String MONGO_AUTH = "?authSource=admin&authMechanism=SCRAM-SHA-1";
    public static final int MONGO_PORT = 27017;
    public static final String DATABASE_NAME = "publisherAPI";

    public static final Map<String, String> MONGO_ENV_VALUES = ImmutableMap.of(
        "MONGO_INITDB_ROOT_USERNAME", MONGO_USER,
        "MONGO_INITDB_ROOT_PASSWORD", MONGO_PASSWORD);

    public static String buildMongoURI(String host, int port, boolean withDbName) {
        String uri = "mongodb://" + MONGO_USER + ":" + MONGO_PASSWORD + "@" + host + ":" + port + "/";

        if(withDbName) {
            uri += DATABASE_NAME;
        }

        return uri + MONGO_AUTH;
    }

    public static final Publisher PUBLISHER_0 = createPublisher("name", "1234");
    public static final Publisher PUBLISHER_1 = createPublisher("test", "3456");
    public static final Publisher PUBLISHER_2 = createPublisher("name2", "abc");
    public static final Publisher PUBLISHER_3 = createPublisher("toBeUpdated", "qwerty");

    public static List<Publisher> PUBLISHERS = Arrays.asList(PUBLISHER_0, PUBLISHER_1, PUBLISHER_2);
    public static List<Publisher> EMPTY_PUBLISHERS = Collections.emptyList();

    public static PublisherDB PUBLISHER_DB = createPublisherDB();

    private static PublisherDB createPublisherDB() {
        PublisherDB newPublisherDB = new PublisherDB();

        newPublisherDB.setId(new ObjectId());
        newPublisherDB.setName("name");
        newPublisherDB.setUri("uri");
        newPublisherDB.setOrganizationId("orgId");
        newPublisherDB.setOrgPath("orgPath");

        PrefLabel prefLabel = new PrefLabel();
        prefLabel.setNb("nbLabel");
        prefLabel.setNn("nnLabel");
        prefLabel.setEn("enLabel");
        newPublisherDB.setPrefLabel(prefLabel);

        return newPublisherDB;
    }

    private static Publisher createPublisher(String name, String orgId) {
        Publisher publisher = new Publisher();

        publisher.setName(name);
        publisher.setOrganizationId(orgId);
        publisher.setOrgPath("orgPath");
        publisher.setUri("uri");

        PrefLabel labels = new PrefLabel();
        labels.setNb("labelNB");
        labels.setNn("labelNN");
        labels.setEn("labelEN");
        publisher.setPrefLabel(labels);

        return publisher;
    }
}

File: src/test/kotlin/no/publishers/TestData.kt
package no.publishers

import com.google.common.collect.ImmutableMap
import no.publishers.generated.model.PrefLabel
import no.publishers.generated.model.Publisher
import no.publishers.model.PublisherDB
import org.bson.types.ObjectId
import java.time.LocalDate

private const val MONGO_USER = "testuser"
private const val MONGO_PASSWORD = "testpassword"
private const val MONGO_AUTH = "?authSource=admin&authMechanism=SCRAM-SHA-1"
const val MONGO_PORT = 27017
const val DATABASE_NAME = "publisherAPI"

val MONGO_ENV_VALUES: Map<String, String> = ImmutableMap.of(
    "MONGO_INITDB_ROOT_USERNAME", MONGO_USER,
    "MONGO_INITDB_ROOT_PASSWORD", MONGO_PASSWORD)

fun buildMongoURI(host: String, port: Int, withDbName: Boolean): String {
    var uri = "mongodb://$MONGO_USER:$MONGO_PASSWORD@$host:$port/"

    if (withDbName) {
        uri += DATABASE_NAME
    }

    return uri + MONGO_AUTH
}

val PUBLISHER_0 = Publisher().apply {
    id = "5d5531e55c404500068481da"
    name = "REGISTERENHETEN I BRØNNØYSUND"
    uri = "http://data.brreg.no/enhetsregisteret/enhet/974760673"
    organizationId = "974760673"
    orgType = "ORGL"
    orgPath = "/STAT/912660680/974760673"
    subOrganizationOf = "http://data.brreg.no/enhetsregisteret/enhet/912660680"
    issued = LocalDate.of(1999, 2, 3)
    uriMunicipalityNumber = "http://www.test.no/fest/1813"
    uriIndustryCode = "http://www.ssb.no/nace/sn2007/84.110"
    uriSectorCode = "http://www.brreg.no/sektorkode/6100"
    prefLabel = PrefLabel().apply {
        nb = "Brønnøysundregistrene"
    }
}

val PUBLISHER_1 =  Publisher().apply {
    id = "5d5531e45c40450006848160"
    name = "ATB AS"
    uri = "http://data.brreg.no/enhetsregisteret/enhet/994686011"
    organizationId = "994686011"
    orgType = "AS"
    orgPath = "/PRIVAT/994686011"
    issued = LocalDate.of(1999, 2, 3)
    uriMunicipalityNumber = "http://www.test.no/fest/5001"
    uriIndustryCode = "http://www.ssb.no/nace/sn2007/84.130"
    uriSectorCode = "http://www.brreg.no/sektorkode/6500"
    prefLabel = PrefLabel().apply {
        nn = "AtB AS"
    }
}

val PUBLISHER_2 =  Publisher().apply {
    id = "5d5531e45c40450006848159"
    name = "FORSVARET"
    uri = "http://data.brreg.no/enhetsregisteret/enhet/986105174"
    organizationId = "986105174"
    orgType = "ORGL"
    orgPath = "/STAT/972417823/986105174"
    subOrganizationOf = "http://data.brreg.no/enhetsregisteret/enhet/972417823"
    uriMunicipalityNumber = "http://www.test.no/fest/0301"
    uriIndustryCode = "http://www.ssb.no/nace/sn2007/84.220"
    uriSectorCode = "http://www.brreg.no/sektorkode/6100"
    prefLabel = PrefLabel().apply {
        en = "Forsvaret"
    }
}

val NEW_PUBLISHER_0 =  Publisher().apply {
    name = "toBeUpdated0"
    uri = "uri0"
    organizationId = "orgId0"
    orgType = "orgForm0"
    orgPath = "/STAT/123/456"
    subOrganizationOf = "http://data.brreg.no/enhetsregisteret/enhet/98765421"
    issued = LocalDate.of(1999, 2, 3)
    uriMunicipalityNumber = "http://www.test.no/fest/0456"
    uriIndustryCode = "industryUri0"
    uriSectorCode = "sectorUri0"
    prefLabel = PrefLabel().apply {
        nn = "nnLabel"
        en = "enLabel"
    }
}

val NEW_PUBLISHER_1 =  Publisher().apply {
    name = "Name"
    uri = "uri1"
    organizationId = "orgId1"
    orgType = "orgForm1"
    orgPath = "/STAT/654/321"
    subOrganizationOf = "http://data.brreg.no/enhetsregisteret/enhet/972417823"
    issued = LocalDate.of(1999, 3, 2)
    uriMunicipalityNumber = "http://www.test.no/fest/1289"
    uriIndustryCode = "industryUri1"
    uriSectorCode = "sectorUri1"
}

val UPDATE_PUBLISHER =  Publisher().apply {
    uri = "uriUpdated"
    organizationId = "orgIdUpdated"
    orgType = "orgFormUpdated"
    orgPath = "/STAT/654/655"
    subOrganizationOf = "http://data.brreg.no/enhetsregisteret/enhet/12345678"
    issued = LocalDate.of(2001, 1, 3)
    uriMunicipalityNumber = "http://www.test.no/fest/6548"
    uriIndustryCode = "industryUriUpdated"
    uriSectorCode = "sectorUriUpdated"
    prefLabel = PrefLabel().apply {
        nb = "nbLabelUpdated"
        nn = "nnLabelUpdated"
        en = "enLabelUpdated"
    }
}

var PUBLISHERS = listOf(PUBLISHER_0, PUBLISHER_1, PUBLISHER_2)
var EMPTY_PUBLISHERS = emptyList<Publisher>()

var PUBLISHER_DB_0 = PublisherDB().apply {
    id = ObjectId("5d5531e55c404500068481da")
    name = "REGISTERENHETEN I BRØNNØYSUND"
    uri = "http://data.brreg.no/enhetsregisteret/enhet/974760673"
    organizationId = "974760673"
    orgType = "ORGL"
    orgPath = "/STAT/912660680/974760673"
    subOrganizationOf = "http://data.brreg.no/enhetsregisteret/enhet/912660680"
    issued = LocalDate.of(1999, 2, 3)
    uriMunicipalityNumber = "http://www.test.no/fest/1813"
    uriIndustryCode = "http://www.ssb.no/nace/sn2007/84.110"
    uriSectorCode = "http://www.brreg.no/sektorkode/6100"
    prefLabel = PrefLabel().apply {
        nb = "Brønnøysundregistrene"
    }
}

val PUBLISHER_DB_1 =  PublisherDB().apply {
    id = ObjectId("5d5531e45c40450006848160")
    name = "ATB AS"
    uri = "http://data.brreg.no/enhetsregisteret/enhet/994686011"
    organizationId = "994686011"
    orgType = "AS"
    orgPath = "/PRIVAT/994686011"
    issued = LocalDate.of(1999, 2, 3)
    uriMunicipalityNumber = "http://www.test.no/fest/5001"
    uriIndustryCode = "http://www.ssb.no/nace/sn2007/84.130"
    uriSectorCode = "http://www.brreg.no/sektorkode/6500"
    prefLabel = PrefLabel().apply {
        nn = "AtB AS"
    }
}

val PUBLISHER_DB_2 =  PublisherDB().apply {
    id = ObjectId("5d5531e45c40450006848159")
    name = "FORSVARET"
    uri = "http://data.brreg.no/enhetsregisteret/enhet/986105174"
    organizationId = "986105174"
    orgType = "ORGL"
    orgPath = "/STAT/972417823/986105174"
    subOrganizationOf = "http://data.brreg.no/enhetsregisteret/enhet/972417823"
    uriMunicipalityNumber = "http://www.test.no/fest/0301"
    uriIndustryCode = "http://www.ssb.no/nace/sn2007/84.220"
    uriSectorCode = "http://www.brreg.no/sektorkode/6100"
    prefLabel = PrefLabel().apply {
        en = "Forsvaret"
    }
}
